### PR TITLE
Add score & action verification (Fixes #4)

### LINKED ISSUE #4: Plugin will not fully work for reCAPTCHA v3
Hello.

Thanks for gem, but after investigation I found it really will not work as expected and only check that reCAPTCHA works inside rails application. Why?

> reCAPTCHA v3 returns a score for each request without user friction. The score is based on interactions with your site and enables you to take an appropriate action for your site

Next move to "Site Verify Response" inside [doc](https://developers.google.com/recaptcha/docs/v3):

```js
{
  "success": true|false,      // whether this request was a valid reCAPTCHA token for your site
  "score": number             // the score for this request (0.0 - 1.0)
  "action": string            // the action name for this request (important to verify)
  "challenge_ts": timestamp,  // timestamp of the challenge load (ISO format yyyy-MM-dd'T'HH:mm:ssZZ)
  "hostname": string,         // the hostname of the site where the reCAPTCHA was solved
  "error-codes": [...]        // optional
}
``` 

So as we can see, `success` only mean that token is valid, nothing more. To check bot or human we need to check `score`, which between 0 and 1. Developer need to adjust inside codebase check for this score based on action (for example `valid?(token, scope)`). 

Also we should for security check `action`, so hacker will not replace on frontend action to some another with lower score and pass verification.

Right now https://github.com/igorkasyanchuk/new_google_recaptcha/blob/master/lib/new_google_recaptcha/validator.rb#L9 only check that token is valid. Library will pass everyone with valid token.

Hope this will help to improve gem.

### PR DESCRIPTION
**Pull request by rubyconvict**

### CODE REVIEW

File: lib/new_google_recaptcha.rb [hunk 1]
```diff
@@ -8,8 +8,8 @@ def self.setup
     yield(self)
   end
 
-  def self.human?(token, model = nil)
-    is_valid = NewGoogleRecaptcha::Validator.valid?(token)
+  def self.human?(token, minimum_score, action, model = nil)
```
Comment (igorkasyanchuk) [inline lib/new_google_recaptcha.rb]: a few questions: 
1) what about having a default value for minimum_score and having it configurable in an initializer?
2) why we need "action"? this is the same action as we specified in view?

File: new_google_recaptcha.gemspec [hunk 2]
```diff
@@ -7,7 +7,7 @@ require "new_google_recaptcha/version"
 Gem::Specification.new do |s|
   s.name        = "new_google_recaptcha"
   s.version     = NewGoogleRecaptcha::VERSION
-  s.authors     = ["Igor Kasyanchuk"]
+  s.authors     = ["Igor Kasyanchuk", "rubyconvict"]
```
Comment (igorkasyanchuk) [inline new_google_recaptcha.gemspec]: I'm 100% agree, you did huge work

also please add yourself into "Contrinutors" section in readme

File: lib/new_google_recaptcha.rb (hunk 1 above)
Comment (igorkasyanchuk) [reply lib/new_google_recaptcha.rb]: 3. we need to have all this documented in readme

Comment (igorkasyanchuk) [reply lib/new_google_recaptcha.rb]: For 2 just update readme, so it will be clear how it works.
Will wait for an updated PR and than merge

**Comment by rubyconvict** [comment]
@igorkasyanchuk Could you merge?

**Comment by igorkasyanchuk** [comment]
@rubyconvict please check and tell if it works, 1.0.0 pushed to rubygems repo

**Comment by rubyconvict** [comment]
@igorkasyanchuk Thanks, it works great!

**Pull request merged**